server/src/main/java/org/elasticsearch/index/mapper/DocumentMapper.java

```diff
@@ -99,19 +99,17 @@ public Builder put(MetadataFieldMapper.Builder mapper) {
             return this;
         }
 
-        public DocumentMapper build(MapperService mapperService) {
+        public DocumentMapper build(IndexSettings indexSettings, DocumentMapperParser documentMapperParser, IndexAnalyzers indexAnalyzers) {
```

Comment:
I don't think we need to change this method signature? Particularly now that the `DocumentMapper` constructor is private, just passing the MapperService here is a much simpler API.

Reply:
I think it depends on perspective. Does the lower number of arguments make it a simpler API? To me, three arguments make it clear what is needed, while `MapperService` means "I may or may not need a bunch of stuff". I think with the three arguments, it is clearer what the method actually uses and needs. `MapperService` is such a big object to carry around.

Reply:
Fair enough - I think we'll end up removing the IndexAnalyzers param soon anyway, so that drops it down to just two parameters which doesn't seem excessive.